src/resolver.ts

```diff
@@ -763,7 +768,7 @@ export class Resolver extends DiagnosticEmitter {
           }
           let defaultType = typeParameterNode.defaultType;
           if (defaultType) {
-            let resolvedDefaultType = this.resolveType(defaultType, ctxFlow.actualFunction, contextualTypeArguments, reportMode);
+            let resolvedDefaultType = this.resolveType(defaultType, prototype, contextualTypeArguments, reportMode);
```

Comment:
This is somewhat similar. The `actualFunction` should be an instance of `prototype`, then sharing state, or is it different?

Reply:
actualFunction is forcaller side.